include/boost/container/small_vector.hpp

```diff
@@ -414,12 +414,12 @@ struct small_vector_storage_calculator_helper<Needed, Hdr, SSize, true>
    static const std::size_t value = 0u;
 };
 
-template<class Storage, class Allocator, class T, std::size_t N>
+template<class Storage, class Allocator, class T, std::size_t N, std::size_t alignment>
 struct small_vector_storage_calculator
 {
-   typedef small_vector_base<T, Allocator> svh_type;
-   typedef vector<T, small_vector_allocator<Allocator> > svhb_type;
-   static const std::size_t s_align = container_detail::alignment_of<Storage>::value;
+   typedef small_vector_base<T, Allocator, alignment> svh_type;
+   typedef vector<T, small_vector_allocator<Allocator, alignment> > svhb_type;
+   static const std::size_t s_align = alignment > 0 ? alignment : container_detail::alignment_of<Storage>::value;
```

Comment:
With `alignment` defaulting to 1, this expression should just also be `static const std::size_t s_align = max<alignment, alignment_of<Storage> >::value;` or similar.